src/components/GalleryBrick/index.js

```diff
@@ -0,0 +1,110 @@
+import React, { Component } from 'react';
+import FontAwesome from 'react-fontawesome';
+import styles from './styles';
+
+class GalleryBrick extends Component {
+  constructor(props){
+    super(props);
+    this.state = { currentPage: 1 }
+  }
+
+  calculatePages(){
+    let { items , itemsPerPage } = this.props;
+    let pages = parseInt(items.length / itemsPerPage);
+
+    return (items.length % itemsPerPage > 0) ? pages + 1 : pages;
+  }
+
+  renderItems(item, key, s){
+    return (
+      <div style={ s.imgGallery }>
+        <img key={ key } style={ s.image} src={ item } />
+      </div>
+      )
+  }
+
+  renderPagesItems(items, s){
+    if (!items) return null;
+
+    let initial = (this.state.currentPage * this.props.itemsPerPage) - this.props.itemsPerPage;
+
+    let boxItems = items.slice(initial, this.props.itemsPerPage * this.state.currentPage );
+
+    return boxItems.map((item, key) => this.renderItems(item, key, s));
+
+  }
+
+  renderPrevBtn(s){
+    if(this.state.currentPage === 1) return null;
+    return (
+      <a 
+        style={ s.controls } 
+        onClick={ () => this.setState({ currentPage : this.state.currentPage - 1 }) } >
+        <FontAwesome name="chevron-left" />
+      </a>
+    )
+  }
+
+  renderNextBtn(s){
+    if(this.state.currentPage >= this.calculatePages()) return null;
+    return (
+      <a 
+        style={ s.controls } 
+        onClick={ () => this.setState({ currentPage : this.state.currentPage + 1 }) } >
+        <FontAwesome name="chevron-right" />
+      </a>
+    )
+  }
+
+  renderIndicators(s){
+    let {currentPage}  = this.state; 
+    let count = [...Array(this.calculatePages()).keys()];
+
+    if(this.calculatePages() < 2) return null;
+
+    return count.map((item) =>{
+      return (
+        <FontAwesome 
+          style={ s.indicator } 
+          onClick={ ()=> this.setState({ currentPage : item + 1 }) } 
+          name={ (currentPage === item + 1) ? 'circle' : 'circle-o' } 
+        />
+      )
+    });
+  }
+
+  render() {
+    let s = styles(this.props);
+
+    let { items, title, description } = this.props;
+    
+    return (
+      <div style={ s.container }>
+        <div style={ s.center }>
+          { (title) ? <h3 style={ s.title }>{ title }</h3> : null }
+          { (description) ? <p style={ s.description } >{ description }</p> : null } 
+        </div>
```

Comment:
Instead of coding the render for the title and description, use a GenericBrick inside this component